### ISSUE TITLE
AnyCPU Class libraries are broken again

### ISSUE THREAD

**Issue by dotMorten** [microsoft/WindowsAppSDK#1217]
**Describe the bug**

Internal bug https://microsoft.visualstudio.com/OS/_workitems/edit/36498869

When building an AnyCPU class library, the build is broken [again](https://github.com/microsoft/WindowsAppSDK/issues/939). Since it's not an application, the architecture runtime limitation shouldn't be in effect.
The issue appears to be due to an incorrect condition in `microsoft.windowsappsdk.foundation\1.0.0-experimental1\build\Microsoft.WindowsAppSDK.Bootstrap.targets`:

```xml
  <Target Name="BinPlaceBootstrapDll" Condition="'$(AppxPackage)' != 'true'" AfterTargets="Build">
    <Copy SourceFiles="$(MSBuildThisFileDirectory)..\runtimes\lib\native\$(_WindowsAppSDKFoundationPlatform)\Microsoft.WindowsAppSDK.Bootstrap.dll" DestinationFolder="$(OutDir)" OverwriteReadOnlyFiles="true" ContinueOnError="false"/>
  </Target>
```

A class library isn't an AppxPackage, but it isn't a running application either.

**Steps to reproduce the bug**

Compile a class lib using reunion:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net5.0-windows10.0.18362.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
    <Configurations>Debug;Release</Configurations>
    <Platforms>AnyCPU</Platforms>
  </PropertyGroup>

  <ItemGroup>
     <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-experimental1" />
  </ItemGroup>
  
</Project>
```

Observe the error:

>1>C:\Users\mn\.nuget\packages\microsoft.windowsappsdk.foundation\1.0.0-experimental1\build\Microsoft.WindowsAppSDK.Bootstrap.targets(11,5): error MSB3030: Could not copy the file "C:\Users\mn\.nuget\packages\microsoft.windowsappsdk.foundation\1.0.0-experimental1\build\..\runtimes\lib\native\AnyCPU\Microsoft.WindowsAppSDK.Bootstrap.dll" because it was not found.

I managed to fix it locally by editing the above condition in the nuget package to this:
  `  Condition="'$(AppxPackage)' != 'true' AND !('$(_WindowsAppSDKFoundationPlatform)' == 'AnyCPU' AND '$(OutputType)'=='Library') "   `

**Expected behavior**
Application compiles without error

**Version Info**

NuGet package version: 
1.0.0-experimental1

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |

**Additional context**

**Comment by andrewleader** [comment]
@AdamBraden is looking into this

**Comment by azchohfi** [comment]
This is blocking:
https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/4171

**Comment by AdamBraden** [comment]
Apologies, the bootstrapper target should not get invoked on dll projects. 
Also when it does run, it shouldn't be binplacing anything under an AnyCPU folder.
@DrusTheAxe as fyi

**Comment by mattleibow** [comment]
We can also remove it for Exe that is MSIX packages. We don't need it.

**Comment by mattleibow** [comment]
I was playing around with all the things, and what I can see the reason for the bootstrap target is to copy the file into the bin directory.

However, this is only needed because the dll is in the wrong folder in the nuget:

Basically instead of:

    runtimes\lib\native\x64\Microsoft.ProjectReunion.Bootstrap.dll

It is now:

    runtimes\win-x64\native\Microsoft.ProjectReunion.Bootstrap.dll

And the BinPlaceBootstrapDll target can go away entirely.

**Comment by AdamBraden** [comment]
@DrusTheAxe - to fully fix this, need update the targets so the bootstrapper DLL is only copied for unpackaged apps and not dll projects. I don't think we can rely upon default nuget behavior, we will still need the targets.

**Comment by dotMorten** [comment]
@AdamBraden What am I missing here? It's the target file you include that makes the mistake, not a default nuget behavior.

**Comment by mattleibow** [comment]
The files in the native folder are only copied for applications during a publish.

So if you make a library or just run, then this file is not copied. Running from the IDE without a publish results in the .NET runtime just referencing the dll from the nuget cache.

**Comment by dotMorten** [comment]
@mattleibow that's not what I'm seeing. I see the entire runtimes folder getting deployed from nuget packages. .NET 5 has default native load behavior to look in that folder due to the dependencies file where they are listed.

**Comment by mattleibow** [comment]
@dotMorten I had a check and got some results:
 - class libraries: not copied
 - apps: copied
 - unit tests: copied

Not sure why I thought that the runtimes were not copied for apps. I swear I had issues in SkiaSharp where people were saying that they wanted the runtimes to be copied to the output, but I guess it might have been for net47 or something.

However, for class libraries, they are not copied. I think that is what @AdamBraden was worried about?

**Comment by dotMorten** [comment]
Right, if just relying on NuGet's `/runtimes/*` folder behavior it won't, since no runtimes are needed for class libs, The current package however uses a .targets file to deploy runtimes and resources, instead of relying on the correct behavior of nuget, and are obviously missing some project type exclusions.
(for .NET Framework we add a .targets file to essentially duplicate the behavior of the net5 target, because as you've found it doesn't apply there, which is a non-issue for the WinAppSDK).

**Comment by dotMorten** [comment]
Temporary workaround I found:
```
  <!-- Workaround for https://github.com/microsoft/WindowsAppSDK/issues/1217 -->
  <Target Name="FixBinPlaceBootstrapDll" BeforeTargets="BinPlaceBootstrapDll" >
    <PropertyGroup>
      <_WindowsAppSDKFoundationPlatform>x86</_WindowsAppSDKFoundationPlatform>
    </PropertyGroup>
  </Target>
```

**Comment by iablaauw-MS** [comment]
@DrusTheAxe should this be marked fixed given your fixes that went into preview2? Or are there still lingering issues here?

**Comment by DrusTheAxe** [comment]
> @DrusTheAxe should this be marked fixed given your fixes that went into preview2? Or are there still lingering issues here?

AnyCPU isn't fixed yet. Here's the set of events to address that https://github.com/microsoft/WindowsAppSDK/pull/1498

**Comment by azchohfi** [comment]
@AdamBraden This seems to have been fixed for 1.0.0 GA, so no workaround needed anymore. Is there any more work needed for this?

**Comment by dotMorten** [comment]
Confirmed!

**Comment by AdamBraden** [comment]
yes, we fixed dlls, but still need to correct Exes because the runtime layout fix didn't make it into 1.0. See #893 and #1777.
